src/loaders/JsonImageInfoLoader.ts

```diff
@@ -233,67 +224,50 @@ class JsonImageInfoLoader implements IVolumeLoader {
-        if (!ctx) {
-          console.log("Error creating canvas 2d context for " + url);
-          return;
-        }
-        ctx.globalCompositeOperation = "copy";
-        ctx.globalAlpha = 1.0;
-        ctx.drawImage(event?.target as CanvasImageSource, 0, 0, w, h);
-        // getImageData returns rgba.
-        // optimize: collapse rgba to single channel arrays
-        const iData = ctx.getImageData(0, 0, w, h);
-
-        const channelsBits: Uint8Array[] = [];
-
-        // allocate channels in batch
-        for (let ch = 0; ch < Math.min(batch.length, 4); ++ch) {
-          channelsBits.push(new Uint8Array(w * h));
-        }
+      const response = await fetch(image.name, { mode: "cors" });
+      const blob = await response.blob();
+      const bitmap = await createImageBitmap(blob);
 
-        // extract the data
-        for (let j = 0; j < Math.min(batch.length, 4); ++j) {
-          for (let px = 0; px < w * h; px++) {
-            channelsBits[j][px] = iData.data[px * 4 + j];
-          }
-        }
+      const canvas = new OffscreenCanvas(bitmap.width, bitmap.height);
+      const ctx = canvas.getContext("2d") as OffscreenCanvasRenderingContext2D | null;
+      if (!ctx) {
+        console.log("Error creating canvas 2d context for " + image.name);
+        return;
+      }
+      ctx.globalCompositeOperation = "copy";
+      ctx.globalAlpha = 1.0;
+      ctx.drawImage(bitmap, 0, 0);
+      const iData = ctx.getImageData(0, 0, bitmap.width, bitmap.height);
+
+      const channelsBits: Uint8Array[] = [];
+      const length = bitmap.width * bitmap.height;
 
-        // done with img, iData, and canvas now.
+      // allocate channels in batch
+      for (let ch = 0; ch < Math.min(image.channels.length, 4); ++ch) {
+        channelsBits.push(new Uint8Array(length));
+      }
 
-        for (let ch = 0; ch < Math.min(batch.length, 4); ++ch) {
-          const chindex = batch[ch];
-          volume.setChannelDataFromAtlas(chindex, channelsBits[ch], w, h);
-          cache?.insert(`${url}/${chindex}`, volume.channels[chindex].volumeData.buffer);
-          onChannelLoaded?.(volume, chindex);
+      // extract the data
+      for (let j = 0; j < Math.min(image.channels.length, 4); ++j) {
+        for (let px = 0; px < length; px++) {
+          channelsBits[j][px] = iData.data[px * 4 + j];
         }
-      };
-      img.crossOrigin = "Anonymous";
-      img.src = url;
-      requests[url] = img;
-    }
+      }
+
+      // done with `iData` and `canvas` now.
+
+      for (let ch = 0; ch < Math.min(image.channels.length, 4); ++ch) {
+        const chindex = image.channels[ch];
+        cache?.insert(`${image.name}/${chindex}`, channelsBits[ch]);
+        // NOTE: the atlas dimensions passed in here are currently unused by `JSONImageInfoLoader`
+        onData(chindex, channelsBits[ch], [bitmap.width, bitmap.height]);
+      }
+    });
 
-    return requests;
+    await Promise.all(loadPromises);
```

Comment:
is this forcing us to block on all channels?  I guess the onData calls will still happen asynchronously but still I wonder if we need to await here?

Reply:
We don't need to await here, in fact I don't think this function is ever await-ed by its callers so this has no impact on control flow. It just seemed odd to produce an array of promises and do nothing with them (though I guess the `map` above could be a `forEach` instead?)